source/val/validate_decorations.cpp

```diff
@@ -864,7 +864,70 @@ spv_result_t CheckVulkanMemoryModelDeprecatedDecorations(
       }
     }
   }
+  return SPV_SUCCESS;
+}
+
+spv_result_t CheckDecorationsOfConversions(ValidationState_t& vstate) {
+  for (const auto& kv : vstate.id_decorations()) {
+    const uint32_t id = kv.first;
+    const auto& decorations = kv.second;
+    if (decorations.empty()) {
+      continue;
+    }
+
+    const Instruction* inst = vstate.FindDef(id);
+    assert(inst);
+
+    // Validates FPRoundingMode decoration
+    for (const auto& decoration : decorations) {
+      if (decoration.dec_type() != SpvDecorationFPRoundingMode) {
+        continue;
+      }
+
+      // Validates width-only conversion instruction for floating-point object
+      // i.e., OpFConvert
+      if (inst->opcode() != SpvOpFConvert) {
+        return vstate.diag(SPV_ERROR_INVALID_ID, inst)
+               << "FPRoundingMode decoration can be applied only to a "
+                  "width-only conversion instruction for floating-point "
+                  "object.";
+      }
+
+      // Validates Object operand of an OpStore
+      for (const auto& use : inst->uses()) {
+        const auto store = use.first;
+        if (store->opcode() != SpvOpStore) {
+            return vstate.diag(SPV_ERROR_INVALID_ID, inst)
+                   << "FPRoundingMode decoration can be applied only to the "
+                      "Object operand of an OpStore.";
+        }
+
+        const auto object_id = store->GetOperandAs<uint32_t>(1);
+        if (object_id != inst->id()) {
```

Comment:
I would change this to use the second element of the use. This would avoid missing some bizarre state where we don't otherwise catch `OpStore %conv %conv`.

Reply:
Updated.